base/dcp_sharded.go

```diff
@@ -118,12 +124,101 @@ func createCBGTIndex(manager *cbgt.Manager, dbName string, bucket Bucket, spec B
 	indexName := dbName + "_import"
 
 	// Register bucketDataSource callback for new index if we need to configure TLS
-	if spec.IsTLS() {
-		cbgt.RegisterBucketDataSourceOptionsCallback(indexName, manager.UUID(), func(options *cbdatasource.BucketDataSourceOptions) *cbdatasource.BucketDataSourceOptions {
-			options.TLSConfig = spec.TLSConfig
-			return options
-		})
-	}
+	cbgt.RegisterBucketDataSourceOptionsCallback(indexName, manager.UUID(), func(options *cbdatasource.BucketDataSourceOptions) *cbdatasource.BucketDataSourceOptions {
+		// A lookup of host dest to external alternate address hostnames
+		externalAlternateAddresses := make(map[string]string)
+		externalAlternateAddressesLock := sync.RWMutex{}
```

Comment:
Does the scoping prevent reusing the code for the non-sharded case here?  Perhaps a helper function that returns ConnectFunc and ConnectBucketFunc?

Reply:
There shouldn't be anything stopping me from sharing this code. I was just being lazy for the POC.